common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -775,6 +775,35 @@ def index_dictionary(self):
 
         return xblock_body
 
+    def get_children_discussion_switches(self) -> dict:
+        """
+        Return dict{child:child.enable_dicussion}
+        """
+        return {
+            child: child.enable_discussion
+            for child in self.get_children()
+        }
+
+    @property
+    def enable_discussion(self):
+        return all(self.get_children_discussion_switches().values())
```

Comment:
As I mentioned before, this is not a true of false value!
We need three options, for, None, All, and Some. i.e.
- None of the children have the flag
- Some of the children have the flag
- All of the children have the flag.